ext/opentelemetry-ext-flask/tests/test_automatic.py

```diff
@@ -0,0 +1,98 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import NOTSET, WARNING, disable
+
+import flask
+from werkzeug.test import Client
+from werkzeug.wrappers import BaseResponse
+
+from opentelemetry import trace
+from opentelemetry.configuration import Configuration
+from opentelemetry.ext.flask import FlaskInstrumentor
+from opentelemetry.test.wsgitestutil import WsgiTestBase
+
+# pylint: disable=import-error
+from .base_test import InstrumentationTest, expected_attributes
+
+
+class TestAutomatic(WsgiTestBase, InstrumentationTest):
+    def setUp(self):
+        super().setUp()
+
+        Configuration._instance = None  # pylint: disable=protected-access
+        Configuration.__slots__ = []  # pylint: disable=protected-access
+        FlaskInstrumentor().instrument()
+
+        self.app = flask.Flask(__name__)
+
+        def hello_endpoint(helloid):
+            if helloid == 500:
+                raise ValueError(":-(")
+            return "Hello: " + str(helloid)
+
+        def excluded_endpoint():
+            return "excluded"
+
+        def excluded2_endpoint():
+            return "excluded2"
+
+        self.app.route("/hello/<int:helloid>")(hello_endpoint)
+        self.app.route("/excluded/<int:helloid>")(hello_endpoint)
+        self.app.route("/excluded")(excluded_endpoint)
+        self.app.route("/excluded2")(excluded2_endpoint)
+
+        self.client = Client(self.app, BaseResponse)
```

Comment:
All this code is duplicated in test_programmatic. What about reducing this by creating a method in `InstrumentationTest` that performs this initialization? or a similar approach?

Reply:
Fixed